workspaces/tests/batch_tx.rs

```diff
@@ -0,0 +1,41 @@
+use serde_json::json;
+use test_log::test;
+use workspaces::prelude::*;
+use workspaces::Function;
+
+#[test(tokio::test)]
+async fn test_batch_tx() -> anyhow::Result<()> {
+    let worker = workspaces::sandbox();
+    let contract = worker
+        .dev_deploy(include_bytes!("../../examples/res/status_message.wasm"))
+        .await?;
+
+    // Batch transaction with two `call`s into `set_status`. The second one
+    // should override the first one.
+    contract
+        .batch(&worker)
+        .call(
+            Function::new("set_status")
+                .args_json(json!({
+                    "message": "hello_world",
+                }))?
+                .deposit(0),
+        )
+        .call(Function::new("set_status").args_json(json!({
+            "message": "world_hello",
+        }))?)
```

Comment:
```suggestion
        .call("set_status")
        .args_json(json!({
            "message": "hello_world",
         }))?
         .deposit(0)
        .call("set_status")
        .args_json(json!({
            "message": "world_hello",
        }))?
```

This type of pattern seems more intuitive for me, but I can see that it could create a more complex builder type (needing to have an in progress function with batch and not) but also doesn't seem that much more complex than having the `Function` builder type having to be built manually.

Is this something you've explored and decided against? Would it be helpful if I made a commit with this change to compare, or do you understand what my suggestion is?

Reply:
Yeah I explored this one already, and there's one part to this that didn't make sense: the above builder syntax would violate type state pattern where we're allowing "non-call" functions to be used with `call` specifying arguments like so:
```rust
account.batch(&worker)
     .create_account()
     .deposit(...)
     .transact()
```
which could be confusing since users might think `deposit` is setting the deposit/transfer-amount into the `create_account` call. So went against this pattern for that reason

Reply:
mmm, I see. So you mean something like this: 
```rust
account
   .batch(&worker)
   .create_account()
   .transfer("1 N")
   .deploy("path/wasm")
   .call("x")
   .args_json((1, 2, 3))
   .deposit(1000)
   .stake("0.5 N")
   .call("y")
   .args_json((4, 5, 6))
   .transact()
```
That could work, but now it might feel like we're breaking the flow with mixing the call argument functions in with the rest of the transaction functions. I like it overall, but still feel a little hesitant with how it flows. What do you think? Or which would you prefer, the above or the following:
```rust
account
   .batch(&worker)
   .create_account()
   .transfer("1 N")
   .deploy("path/wasm")
   .call(
       Function::new("x")
          .args_json((1, 2, 3))
          .deposit(1000)
    )
   .stake("0.5 N")
   .call(
       Function::new("x")
          .args_json((4, 5, 6))
   )
   .transact()
```

Reply:
To be clear, I'm totally fine with what there was, just wanted to offer an alternative suggestion since it seemed a bit unergonomic to use for a script like this

Reply:
ended up just keeping the former, but we can go back to this if we get enough feedback that doesn't like pulling in `Function` to use with `call`s